app/scripts/services/validateRegistrant.js

```diff
@@ -26,9 +25,9 @@ angular.module('confRegistrationWebApp')
           }
         }
         if ((ruleOperand === 'OR' && validRuleCount > 0) || (ruleOperand === 'AND' && validRuleCount <= i)) {
-          break;
+          return false; // Exit lodash foreach as we found a case which determines the whole outcome of this function
```

Comment:
You wouldn't want to return `false` if `(ruleOperand === 'OR' && validRuleCount > 0)`.

Maybe instead:
```
if (ruleOperand === 'AND' && validRuleCount <= i) {
      return false; // Exit lodash foreach as we found a case which determines the whole outcome of this function
}
if (ruleOperand === 'OR' && validRuleCount > 0) {
      return true; // Exit lodash foreach as we found a case which determines the whole outcome of this function   
}
```